examples/read.c

```diff
@@ -4,19 +4,62 @@
 
 #include "wiringX.h"
 
-int main(void) {
+char *usage = "Usage: %s GPIO GPIO\n"
+			"       first GPIO to write to = output\n"
+			"       second GPIO to read from = input\n"
+			"Example: %s 4 23\n";
+
+int main(int argc, char *argv[]) {
+	int gpio_out, gpio_in;
+	int i, n, value;
+	char ch, str [100];
+
+	// first, check for valid, numeric arguments
+	// found at http://stackoverflow.com/questions/17292545
+	for(i=1; i<argc; i++) {
+		n = sscanf(argv[i], "%d%c", &value, &ch);
```

Comment:
I have played with isdigt but I found is to complicated, especially for me :-(

    char *endptr, *str;

    for(i=1; i<argc; i++) {
       str = argv[i];
       val = (int)strtol(str, &endptr, 10);
       if((*endptr != '\0') || (val < 0 || val >= 128)) {
          printf("%s: Invalid GPIO %s\n", argv[0], str);
          return -1;
       }
    }

Also with these, the following segmentation fault is gone with a GPIO above 127.

    ./read 128 6
    running on a raspberrypi
    Segmentation fault (core dumped)

    ./read 127 6
    running on a raspberrypi
    ./read: Invalid GPIO 127 for output

now:

    ./read 544 22 
    ./read: Invalid GPIO 544

    ./read 11g 22
    ./read: Invalid GPIO 11g